src/core/mac/data_poll_sender.cpp

```diff
@@ -317,6 +317,12 @@ void DataPollSender::CheckFramePending(Mac::RxFrame &aFrame)
     {
         SendDataPoll();
     }
+    else if (aIsAck)
```

Comment:
I think it is better to limit the function `CheckFramePending` to send data poll in accordance with its name. Maybe we can put this keep alive handling to a separate function or just add it in `Mac::HandleReceivedFrame` and `Mac::HandleTransmitDone`?

Reply:
I agree that we should separate keep alive handling from frame pending handling.

Reply:
@abtink suggested reuse the method. And I also found it can avoid duplicate code. I agree to update the name as `ProcessFrame(aFrame)` to better reflect the functionality.